taskcluster/ac_taskgraph/loader/build_config.py

```diff
@@ -13,7 +17,113 @@
 from ..build_config import get_components
 
 
+logger = logging.getLogger(__name__)
+
+ALL_COMPONENTS = object()
+# We always build components that have changed files or dependencies, but we also need
+# to rebuild some or all components when other files change. This structure
+# maintains the latter mapping.
+FILE_PATTERNS_TO_AFFECTED_COMPONENTS = {
+    # Just a placeholder example
+    #"^.*changelog.md.*$": ["build-browser-awesomebar"],
+    "^build.gradle$": ALL_COMPONENTS,
+}
+
+def get_files_changed_pr(base_repository, pull_request_number):
+    url = base_repository.replace("github.com", "api.github.com/repos")
+    url += "/pulls/%s/files" % pull_request_number
+
+    r = requests.get(url, timeout=60)
+    r.raise_for_status()
+    files = [f["filename"] for f in r.json()]
+    return files
+
+def get_files_changed_push(base_repository, base_rev, head_rev):
+    url = base_repository.replace("github.com", "api.github.com/repos")
+    url += "/compare/"
+    url += "%s...%s" % (base_rev, head_rev)
+
+    r = requests.get(url, timeout=60)
+    r.raise_for_status()
+    files = [f["filename"] for f in r.json().get("files")]
+    return files
+
+def get_components_changed(files_changed):
+    """Translate a list of files changed into a list of components. Eg:
+        [
+            "components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/GeckoLoginStorageDelegate.kt",
+            "components/service/sync-logins/src/test/java/mozilla/components/service/sync/logins/GeckoLoginStorageDelegateTest.kt",
+        ]
+        ->
+        {service-sync-logins}
+    """
+    return set(["-".join(f.split("/")[1:3]) for f in files_changed if f.startswith("components")])
+
+
+cached_deps = {}
+
+def get_deps_for_component(component):
+    """Return the full list of local upstream dependencies of a component."""
+    # Gradle is expensive to run. Cache the results for each component.
+    # TODO: replace me with a functools.lru_cache decorator when this runs on py3
+    if component in cached_deps:
+        return cached_deps[component]
+
+    deps = set()
+    # Parsing output like this is not ideal but bhearsum couldn't find a way
```

Comment:
`gradle` allows us to define custom tasks. We made `printVariants` in Fenix, for instance: https://github.com/mozilla-mobile/fenix/blob/667261f511b7fdd7f45e289502fa7a481512b108/app/build.gradle#L660. 

I spent an hour and half on trying to make such a task but I couldn't. @pocmo would you have an idea?

Reply:
I don't have any good ideas here. Like Johan I'd probably spend quite some time trying various things.. :) Maybe we can file this separately as a follow-up and then we can see if we can find a better solution using gradle? :)

Reply:
That sounds good to me. I'll open a new Issue for it.

Reply:
https://github.com/mozilla-mobile/android-components/issues/7814